### ISSUE TITLE
Errors when building with `-pedantic` on gcc 4.9.2

### ISSUE THREAD

**Issue by MacoChris** [google/fruit#26]
Building against the 2.0.4 tag with pedantic causes the error:

```
fruit/include/fruit/impl/binding_data.defn.h:35:54: error: zero-size array ‘types’
     static const TypeId types[] = {getTypeId<Ts>()...};
```

Easiest reproduction is building the hello_world example with `-pedantic` on:

```
diff --git a/examples/hello_world/CMakeLists.txt b/examples/hello_world/CMakeLists.txt
index 5a154ad..6c6d599 100644
--- a/examples/hello_world/CMakeLists.txt
+++ b/examples/hello_world/CMakeLists.txt
@@ -4,4 +4,5 @@ main.cpp
 )

 add_executable(hello ${HELLO_WORLD_SOURCES})
+target_compile_options(hello PRIVATE -pedantic)
 target_link_libraries(hello fruit)
```

The cause seems to be using multiple `.bind` statements when creating a component, like in `examples/hello_world/main.cpp:46`

**Comment by poletti-marco** [comment]
Hi, thanks for the report.
I'll take a look when I have time (hopefully this weekend).
If that's the only issue it seems easy to fix, and it might make Fruit more portable.

**Comment by poletti-marco** [comment]
This was fixed in 8b2abe62e57bb02e00f324e8ffcdb4cfa886461d, and the fix will be part of the next release (Fruit 2.1.0).
